src/include/error.h

```diff
@@ -176,3 +176,15 @@
             WT_RET_PANIC(session, v, __VA_ARGS__); \
     } while (0)
 #endif
+
+/*
+ * WT_ASSERT_STRONG
```

Comment:
I'm open to any naming suggestions here. My intention with this name is that grepping for `WT_ASSERT` will find all assertions in the code.

Reply:
We could go with something like `WT_ASSERT_PROD` or `WT_ASSERT_RELEASE`. I don't really mind `strong` though in this context. I am interested in how we would instruct developers on whether they should use one of the new release build asserts or a panic?

Reply:
Good question.. For the purposes of this review I think we can leave  this to the PM-2897 project. I've added a comment on the (to-be-written) scope document to discuss this.

Reply:
I prefer `WT_ASSERT_ALWAYS` - I think with `_STRONG` I'll need to go and lookup what that means.

Further - per the comment above the set of macros here - the no-message assert macro is supposed to be deprecated. I'd like for this to require a message describing what the assertions are enforcing. Hopefully that's often a matter of moving a comment into an error message.

I was slow in replying, because I'd actually like to switch the default here - to be `WT_ASSERT` is always on, but we have a `WT_ASSERT_DIAGNOSTIC` (or similar) for checks that only apply in diagnostic builds. I think that can wait for another build.

Reply:
I prefer `WT_ASSERT_ALWAYS` - I think with `_STRONG` I'll need to go and lookup what that means.

Further - per the comment above the set of macros here - the no-message assert macro is supposed to be deprecated. I'd like for this to require a message describing what the assertions are enforcing. Hopefully that's often a matter of moving a comment into an error message.

I was slow in replying, because I'd actually like to switch the default here - to be `WT_ASSERT` is always on, but we have a `WT_ASSERT_DIAGNOSTIC` (or similar) for checks that only apply in diagnostic builds. I think that can wait for another build.

Reply:
Sounds good! 
As discussed we'll leave the default WT_ASSERT as diagnostic mode only for now. I've pushed [12b075d](https://github.com/wiredtiger/wiredtiger/pull/8004/commits/12b075d0b30b280abc160a1baf1b6fb9a7d2b228) to update the name to `WT_ASSERT_ALWAYS` and [474fde0](https://github.com/wiredtiger/wiredtiger/pull/8004/commits/474fde0bb3828f3f068e7708afdcdfe55076567e) to add the reason for assertion failures. @luke-pearson @quchenhao would you mind re-reviewing to confirm the new assertion fail reasons match the asserts?